core/the/eventwriter.go

```diff
@@ -25,14 +25,47 @@
 package the
 
 import (
+	"sync"
+
 	"github.com/AliceO2Group/Control/common/event"
 	"github.com/AliceO2Group/Control/common/event/topic"
+	"github.com/AliceO2Group/Control/common/logger"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	writers = make(map[topic.Topic]*event.Writer)
+	mu      sync.Mutex
+	log     = logger.New(logrus.StandardLogger(), "core")
 )
 
+func createOrGetWriter(topic topic.Topic) *event.Writer {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if writer, ok := writers[topic]; ok {
+		return writer
+	}
+
+	writers[topic] = event.NewWriterWithTopic(topic)
+	return writers[topic]
+}
+
 func EventWriter() *event.Writer {
-	return EventWriterWithTopic(topic.Root)
+	return createOrGetWriter(topic.Root)
 }
 
 func EventWriterWithTopic(topic topic.Topic) *event.Writer {
-	return event.NewWriterWithTopic(topic)
+	return createOrGetWriter(topic)
+}
+
+func Clear() {
+	mu.Lock()
+	defer mu.Unlock()
+
+	log.Logf(logrus.InfoLevel, "Clearing %d kafka producers", len(writers))
+	for _, writer := range writers {
+		writer.Close()
+	}
+	clear(writers)
 }
```

Comment:
I would propose to either rename it to `ClearEventWriter` since it does only that. If we want it to clear all the singletons, we can refactor it later, but for now the call:
```
the.Clear()
```
is misleading.

Otherwise, i think we could move on with this PR.